### ISSUE TITLE
Relay vs generic GraphQL

### ISSUE THREAD

**Issue by vpetrovykh** [edgedb/edgedb#1582]
Our GraphQL implementation was already inspired by relay and some of its features. Namely, the `before`, `after`, `first`, `last` spec to get slices of multi-links are similar to relay connections (without the intermediate "edges" structure). This is covered in issue #187.

At this point it seems more productive to differentiate our GraphQL endpoints into _generic_ and _relay-compatible_ flavors. Basically, the _generic_ GraphQL endpoint would attempt to adhere to database concepts and naming (such as `limit` and `offset` for selecting a portion of the results), while the _relay-compatible_ one would implement relay connections with `before`, `after`, `first`, `last` arguments and output edges, nodes and [pageInfo](https://relay.dev/graphql/connections.htm#sec-undefined.PageInfo) as expected.

The objective here is to provide more convenient interfaces to users who use relay as well as less bulky interface to people who don't.

(Implementing two different GraphQL end-points has been inspired by Hasura)

**Comment by vpetrovykh** [comment]
To implement _relay-compatible_ output we can use computables with named tuples and aggregates in place of multi-links. This requires disabling the type-check for overriding computables, but should be fundamentally doable within the scope of GraphQL-to-EdgeQL compilation.

**Comment by 1st1** [comment]
> Namely, the before, after, first, last spec to get slices of multi-links are similar to relay connections

I think we should change this to limit/offset for non-relay mode.

**Comment by vpetrovykh** [comment]
> I think we should change this to limit/offset for non-relay mode.

Yes, I mentioned that and we can make this more explicit here. One simplification is that we want to use `offset` and `limit` instead of the current `before`, `after`, `first`, `last` combo for _generic_ GraphQL endpoint. They are actually far easier to "get right" as `after` is not a perfect equivalent to `offset` and it's easy to make an "off by one" mistake with it if alternating between "after" and array indices (and in-code array indices are likely to be used).

**Comment by trompx** [comment]
Hello,
I am interested in relay cursor pagination and was wondering if you would also consider supporting page cursors similar to the artsy relay pagination implementation (discussed [in this github issue](https://github.com/devoxa/prisma-relay-cursor-connection/pull/366))?

+1 for using offset/limit for non-relay
